miio/g1vacuum.py

```diff
@@ -0,0 +1,281 @@
+    def side_brush_life_level(self) -> int:
+        """Side Brush Life Level in %."""
+        return self.data["side_brush_life_level"]
+
+    @property
+    def filter_life_level(self) -> int:
+        """Filter Life Level in %."""
+        return self.data["filter_life_level"]
+
+    @property
+    def clean_area(self) -> int:
+        """Clean Area in cm2."""
+        return self.data["clean_area"]
+
+    @property
+    def clean_time(self) -> timedelta:
+        """Clean Time in Minutes."""
+        return self.data["clean_time"]
+
+    @property
+    def total_clean_count(self) -> int:
+        """Total Clean Count."""
+        return self.data["total_clean_count"]
+
+
+
+
+class G1Vacuum(MiotDevice):
+    """Support for G1 vacuum (G1, mijia.vacuum.v2)."""
+
+    mapping = MIOT_MAPPING[MIJIA_VACUUM_V2]
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MIJIA_VACUUM_V2,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+        self.model = model
+
+    @command(
+        default_output=format_output(
+            "",
+            "State: {result.state}\n"
+            "Error: {result.error}\n"
+            "Battery: {result.battery}%\n"
+            "Mode: {result.operating_mode}\n"
+            "Mop State: {result.mop_state}\n"
+            "Charge Status: {result.charge_state}\n"
+            "Fan speed: {result.fan_speed}\n"
+            "Water level: {result.water_level}\n"
+            "Filter Life Level: {result.filter_life_level}%\n"
+            "Main Brush Life Level: {result.main_brush_life_level}%\n"
+            "Side Brush Life Level: {result.side_brush_life_level}%\n"
+            "Clean Area: {result.clean_area}\n"
+            "Clean Time: {result.clean_time}\n"
+            "Total Clean Count: {result.total_clean_count}\n",
+        )
+    )
+
+    def status(self) -> G1Status:
+        """Retrieve properties."""
+
+        return G1Status(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+
+    @command()
+    def home(self):
+        """Home."""
+        return self.call_action("home")
+
+    @command()
+    def start(self) -> None:
+        """Start Cleaning"""
+        return self.call_action("start")
+
+    @command()
+    def stop(self):
+        """Stop Cleaning"""
+        return self.call_action("stop")
+
+    @command()
+    def find(self) -> None:
+        """Find the robot."""
+        return self.call_action("find")
+
+    @command(click.argument("consumable", type=G1Consumable))
+    def consumable_reset(self, consumable: G1Consumable):
+        """Reset consumable information. CONSUMABLE=main_brush_life_level|side_brush_life_level|filter_life_level"""
+        if consumable.name == "MainBrush":
```

Comment:
```suggestion
        if consumable == G1Consumable.MainBrush:
```
You can directly use the enums here and below for comparisons.